test/common.js

```diff
@@ -20,33 +20,44 @@ if (process.env.TEST_THREAD_ID) {
 exports.tmpDir = path.join(exports.testDir, exports.tmpDirName);
 
 var opensslCli = null;
+var inFreeBSDJail = null;
+var localhostIPv4 = null;
 
 Object.defineProperty(exports, 'inFreeBSDJail', {
   get: function() {
+    if (inFreeBSDJail !== null) return inFreeBSDJail;
+
     if (process.platform === 'freebsd' &&
       child_process.execSync('sysctl -n security.jail.jailed').toString() ===
       '1\n') {
-      return true;
+      inFreeBSDJail = true;
     } else {
-      return false;
+      inFreeBSDJail = false;
     }
+    return inFreeBSDJail;
   }
 });
 
-Object.defineProperty(exports, 'localhost_ipv4', {
+Object.defineProperty(exports, 'localhostIPv4', {
   get: function() {
+    if (localhostIPv4 !== null) return localhostIPv4;
+
     if (exports.inFreeBSDJail) {
       // Jailed network interfaces are a bit special - since we need to jump
       // through loops, as well as this being an exception case, assume the
       // user will provide this instead.
-      if (process.env.LOCALHOST)
-        return process.env.LOCALHOST;
-
-      console.error('Looks like we\'re in a FreeBSD Jail. ' +
-                    'Please provide your default interface address ' +
-                    'as LOCALHOST or expect some tests to fail.');
+      if (process.env.LOCALHOST) {
+        localhostIPv4 = process.env.LOCALHOST;
+      } else {
+        console.error('Looks like we\'re in a FreeBSD Jail. ' +
+                      'Please provide your default interface address ' +
+                      'as LOCALHOST or expect some tests to fail.');
+      }
+    } else {
+      localhostIPv4 = '127.0.0.1';
```

Comment:
This will never be reached for FreeBSD jails that hasn't set LOCALHOST. I think falling back to '127.0.0.1' is the right thing here.